src/main/scala/org/graphframes/GraphFrame.scala

```diff
@@ -360,6 +361,51 @@ class GraphFrame private(
    */
   def aggregateMessages: AggregateMessages = new AggregateMessages(this)
 
+  /**
+   * Filter the vertices according to Column expression, remove edges containing any dropped
+   * vertices.
+   * @group subgraph
+   */
+  def filterVertices(condition: Column): GraphFrame = {
+    val vv = vertices.filter(condition)
+    val ee = edges.join(vv, vv(ID) === edges(SRC), "left_semi")
+                  .join(vv, vv(ID) === edges(DST), "left_semi")
+    GraphFrame(vv, ee)
+  }
+
+  /**
+   * Filter the vertices according to String expression, remove edges containing any dropped
+   * vertices.
+   * @group subgraph
+   */
+  def filterVertices(conditionExpr: String): GraphFrame = filterVertices(expr(conditionExpr))
+
+  /** 
+   * Filter the edges according to Column expression, keep all vertices.
+   * @group subgraph
+   */
+  def filterEdges(condition: Column): GraphFrame = {
+    val vv = vertices
+    val ee = edges.filter(condition)
+    GraphFrame(vv, ee)
+  }
+
+  /** 
+   * Filter the edges according to String expression.
+   * @group subgraph
+   */
+  def filterEdges(conditionExpr: String): GraphFrame = filterEdges(expr(conditionExpr))
+
+  /**
+   * Drop isolated vertices, vertices not contained in any edges.
+   * @group subgraph
+   */
+  def dropIsolatedVertices(): GraphFrame = {
+    val ee = edges
+    val e1 = ee.withColumn(ID, explode(array(col(SRC), col(DST))))
+    val vv = vertices.join(e1, Seq(ID), "left_semi")
```

Comment:
It will add duplicates. We need to add distinct to e1. Please also add a test.

Reply:
I think the result will not include the duplicates if the vertices do not include duplicates . So we do not need an extra distinct operation ?

Reply:
you are right!